Document developer process for generating school check data (#888)
DFE-Digital/dfe-teachers-payment-service: docs/school-check-data.md

# Generating School Check Data

**NB: This process involves accessing the production console. You will need an
approved PIM request, and a second developer to observe during this.**

> As part of verifying a teacher's claim service operators will send emails to the

Reviewer: ```suggestion
As part of verifying a teacher's claims, service operators will send an email to the
```

> relevant schools to check their employment status. Generating a list of claims

Reviewer: It's not strictly speaking employment status, it's a bit more complicated than that and varies per policy. Let's not be too specific here so as not to mislead:

```suggestion
relevant school to confirm their eligibility. Generating a list of claims
```

which need this email sending is a developer task, documented below.

## Process

1. Visit the
   [School Check Email - Sent Claims spreadsheet](https://docs.google.com/spreadsheets/d/1mxMND-SqOjK7yyYp0EaEJOOCUYFkqLJ1ZIwPLuSVKfM/edit#gid=1926551388)

2. On the second sheet, copy the contents of cell A1. This should be a
   comma-delimited list of all the claim references which have already been
   checked, for example `A1B2C345,D6E7F890,G1H2I345...`. Check to make sure
   there are no trailing commas

3. Have a second developer join you too observe, and connect to the production
   console using

   ```
   bin/azure-console production
   ```

4. Activate the Rails console in sandbox mode with

   ```

>    bundle exec rails console --sandbox

Reviewer: For reference, using the bin file also works with less typing:
```suggestion
   bin/rails console --sandbox
```

   ```

5. Create a new instance of `Claim::SchoolCheckEmailDataExport` with

   ```
   exporter = Claim::SchoolCheckEmailDataExport.new("A1B2C345,D6E7F890,G1H2I345...")
   ```

   substituting the claim references for the string you copied from the

>    spreadsheet

Reviewer: ```suggestion
   spreadsheet in step 2.
```

6. Generate the CSV data of claims needing checking by running

   ```

>    exporter.csv_string

Reviewer: Using `puts` here will directly output the string to the terminal STDOUT making step 7 redundant.

```suggestion
   puts exporter.csv_string
```

   ```

   You may need to press the space bar to make your terminal render to the end
   of the string

> 7. Copy the string into your text editor, and make the following character

Reviewer: Change step 6 above and you can drop this step.

   replacements:

   - Replace `\n` with a line break
   - Replace `\"` with `"`

8. Save the file as a CSV with the name `school-check-batch_ddmmyyyy.csv`

> 9. Send the file to the relevant service operator to follow the rest of the

Reviewer: ```suggestion
9. Securely transfer the file to the relevant service operator to follow the rest of the
```

   school check process

10. From the CSV file, copy the claim references (and no other details) and
    append them to the first worksheet in the

>     [School Check Email - Sent Claims spreadsheet](https://docs.google.com/spreadsheets/d/1mxMND-SqOjK7yyYp0EaEJOOCUYFkqLJ1ZIwPLuSVKfM/edit#gid=1926551388).

Reviewer: ```suggestion
    [School Check Email - Sent Claims spreadsheet](https://docs.google.com/spreadsheets/d/1mxMND-SqOjK7yyYp0EaEJOOCUYFkqLJ1ZIwPLuSVKfM/edit#gid=1926551388) so that they will be excluded the next time the process is run.
```
